src/keywordPreservation.ts

```diff
@@ -213,7 +213,15 @@ export const preserveKeywords = ({
 
   addresses.forEach((address) => {
     const localValue = getAssetsValueByAddress(address, localAssets);
-    const remoteValue = getAssetsValueByAddress(address, remoteAssets);
+
+    const remoteAssetsAddress = (() => {
+      const doesAddressHaveKeyword = doesHaveKeywordMarker(address, keywordMappings);
+      if (doesAddressHaveKeyword) {
+        return keywordReplace(address, keywordMappings);
+      }
+      return address;
```

Comment:
This conditional check isn't completely necessary but it reduces the blast radius of any issues that could occur if an erroneous keyword replacement happened. Also, it serves to document why two codepaths exist.